File: lib/src/model/GetObjectResult.java
package com.alibaba.sdk.android.oss.model;

import com.alibaba.sdk.android.oss.internal.CheckCRC64DownloadInputStream;

import java.io.InputStream;

/**
 * Created by zhouzhuo on 11/23/15.
 */
 class GetObjectResult extends OSSResult {

    // object metadata
     ObjectMetadata metadata = ObjectMetadata();

    // content length
     int contentLength;

    // object's content
     InputStream objectContent;

    /**
     * Gets the metadata
     *
     * @return object metadata
     */
     ObjectMetadata getMetadata() {
        return metadata;
    }

     void setMetadata(ObjectMetadata metadata) {
        this.metadata = metadata;
    }

    /**
     * Gets the object content
     *
     * @return Object's content in the form of InoutStream
     */
     InputStream getObjectContent() {
        return objectContent;
    }

     void setObjectContent(InputStream objectContent) {
        this.objectContent = objectContent;
    }

    /**
     * Gets the object length
     *
     * @return object length
     */
     int getContentLength() {
        return contentLength;
    }

     void setContentLength(int contentLength) {
        this.contentLength = contentLength;
    }

    @override
     int getClientCRC() {
        if (objectContent != null && (objectContent instanceof CheckCRC64DownloadInputStream)) {
            return ((CheckCRC64DownloadInputStream) objectContent).getClientCRC64();
        }
        return super.getClientCRC();
    }
}

File: lib/src/model/get_object_result.dart
import 'package:aliyun_oss_dart_sdk/src/internal/check_crc64_download_input_stream.dart';
import 'package:aliyun_oss_dart_sdk/src/internal/http_message.dart';
import 'object_metadata.dart';
import 'oss_result.dart';

class GetObjectResult extends OSSResult {
  // object metadata
  ObjectMetadata metadata = ObjectMetadata();

  // content length
  int contentLength = 0;

  // object's content
  InputStream? objectContent;

  @override
  String? get clientCRC {
    if (objectContent != null &&
        (objectContent is CheckCRC64DownloadInputStream)) {
      return objectContent.cli();
    }
    return super.clientCRC;
  }
}
